packages/scheduler/src/SchedulerProfiling.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {PriorityLevel} from './SchedulerPriorities';
+import {enableProfiling} from './SchedulerFeatureFlags';
+
+import {NoPriority} from './SchedulerPriorities';
+
+let runIdCounter: number = 0;
+let mainThreadIdCounter: number = 0;
+
+const profilingStateSize = 4;
+export const sharedProfilingBuffer = enableProfiling
+  ? // $FlowFixMe Flow doesn't know about SharedArrayBuffer
+    typeof SharedArrayBuffer === 'function'
+    ? new SharedArrayBuffer(profilingStateSize * Int32Array.BYTES_PER_ELEMENT)
+    : // $FlowFixMe Flow doesn't know about ArrayBuffer
+      typeof ArrayBuffer === 'function'
+      ? new ArrayBuffer(profilingStateSize * Int32Array.BYTES_PER_ELEMENT)
+      : null // Don't crash the init path on IE9
+  : null;
+
+const profilingState =
+  enableProfiling && sharedProfilingBuffer !== null
```

Comment:
Keeping this `sharedProfilingBuffer !== null` check because while `new Int32Array(null)` would work, IE10 which doesn't have `ArrayBuffer` *also* doesn't have `Int32Array`.